cmake/helpers.cmake

```diff
@@ -36,6 +36,11 @@ function(add_umf_target_compile_options name)
         if(UMF_DEVELOPER_MODE)
             target_compile_options(${name} PRIVATE
                 -Werror
+                -Wextra
+                -Wshadow
+                -Wno-implicit-fallthrough
```

Comment:
Do we use a fallthrough switch statement somewhere? I think the code is cleaner with explicit breaks in switches.

Reply:
no, but oneCCL enables this flag so I would leave it but as you suggest - we'll probably use [[fallthrough]] attribute for such cases